sdk/storage/azure-storage-quickquery/src/test/java/com/azure/storage/quickquery/APISpec.groovy

```diff
@@ -0,0 +1,398 @@
+    }
+
+    static BlobLeaseClient createLeaseClient(BlobClientBase blobClient) {
+        return createLeaseClient(blobClient, null)
+    }
+
+    static BlobLeaseClient createLeaseClient(BlobClientBase blobClient, String leaseId) {
+        return new BlobLeaseClientBuilder()
+            .blobClient(blobClient)
+            .leaseId(leaseId)
+            .buildClient()
+    }
+
+    static BlobLeaseClient createLeaseClient(BlobContainerClient containerClient, String leaseId) {
+        return new BlobLeaseClientBuilder()
+            .containerClient(containerClient)
+            .leaseId(leaseId)
+            .buildClient()
+    }
+
+    /**
+     * This will retrieve the etag to be used in testing match conditions. The result will typically be assigned to
+     * the ifMatch condition when testing success and the ifNoneMatch condition when testing failure.
+     *
+     * @param bc
+     *      The URL to the blob to get the etag on.
+     * @param match
+     *      The ETag value for this test. If {@code receivedEtag} is passed, that will signal that the test is expecting
+     *      the blob's actual etag for this test, so it is retrieved.
+     * @return
+     * The appropriate etag value to run the current test.
+     */
+    def setupBlobMatchCondition(BlobClientBase bc, String match) {
+        if (match == receivedEtag) {
+            return bc.getProperties().getETag()
+        } else {
+            return match
+        }
+    }
+
+    def setupBlobMatchCondition(BlobAsyncClientBase bac, String match) {
+        if (match == receivedEtag) {
+            return bac.getProperties().block().getETag()
+        } else {
+            return match
+        }
+    }
+
+    /**
+     * This helper method will acquire a lease on a blob to prepare for testing lease Id. We want to test
+     * against a valid lease in both the success and failure cases to guarantee that the results actually indicate
+     * proper setting of the header. If we pass null, though, we don't want to acquire a lease, as that will interfere
+     * with other AC tests.
+     *
+     * @param bc
+     *      The blob on which to acquire a lease.
+     * @param leaseID
+     *      The signalID. Values should only ever be {@code receivedLeaseID}, {@code garbageLeaseID}, or {@code null}.
+     * @return
+     * The actual lease Id of the blob if recievedLeaseID is passed, otherwise whatever was passed will be
+     * returned.
+     */
+    def setupBlobLeaseCondition(BlobClientBase bc, String leaseID) {
+        String responseLeaseId = null
+        if (leaseID == receivedLeaseID || leaseID == garbageLeaseID) {
+            responseLeaseId = createLeaseClient(bc).acquireLease(-1)
+        }
+        if (leaseID == receivedLeaseID) {
+            return responseLeaseId
+        } else {
+            return leaseID
+        }
+    }
+
+    def setupBlobLeaseCondition(BlobAsyncClientBase bac, String leaseID) {
```

Comment:
If you can, cleaning out some of these helpers that we don't need would be helpful.